.github/workflows/test_frontend_part1.yaml

```diff
@@ -0,0 +1,162 @@
+name: Test Skyportal Front-end Part 1
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  test:
+    name: Test SkyPortal
+    runs-on: ubuntu-latest
+    timeout-minutes: 90
+
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_USER: skyportal
+          POSTGRES_PASSWORD: anything
+        ports:
+          - 5432:5432
+        # needed because the postgres container does not provide a
+        # healthcheck
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    steps:
+      - uses: actions/setup-python@v1
+        with:
+          python-version: "3.8"
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache
+            ~/.local
+            ~/.npm
+            cache
+          key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}
```

Comment:
It may be worth figuring out the correct caching key; I'm not sure our `key` is correct, and I imagine it costs us not to utilize the cache properly.

Reply:
I'd prefer to leave that for another PR, as it's unrelated to this one.